addons/website_sale/views/templates.xml

```diff
@@ -1318,7 +1318,13 @@
     </template>
 
     <template id="product_price">
-        <div itemprop="offers" itemscope="itemscope" itemtype="http://schema.org/Offer" t-attf-class="product_price mt-2 mb-3 {{'d-none' if combination_info['prevent_zero_price_sale'] else 'd-inline-block'}}">
+        <div
+            t-if="not combination_info['prevent_zero_price_sale']"
+            itemprop="offers"
+            itemscope="itemscope"
+            itemtype="http://schema.org/Offer"
+            t-attf-class="product_price mt-2 mb-3 d-inline-block"
+        >
```

Comment:
I think this will also hide rental/subscription prices if the product sales price is 0 :eyes: (and the rental/sub prices aren't).
I believe we should update the overrides of `_get_combination_info` in `website_sale_subscription` and `website_sale_renting` to update `prevent_zero_price_sale` (it seems they never considered it at all :D).

Reply:
Indeed, there is no issue with rental :ok_hand: 
If it's not too complex, I'd say that the same check could be added in both modules to keep both logics similar.